spec/lib/json_schema/validator_adapter_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "json_schema/validator_adapter"
+
+describe JsonSchema::ValidatorAdapter do
+  context "when validating" do
+    let(:body) { { "a" => 5 } }
+    let(:invalid_body) { { "b" => 6 } }
+    let(:extra_attributes) { { "a" => 42, "b" => 16, "extra_stuff" => "not required" } }
+    let(:schema) do
+      {
+        "type" => "object",
+        "required" => %w[a],
+        "properties" => {
+          "a" => { "type" => "integer", "default" => 42 },
+          "b" => { "type" => "integer" },
+        },
+        "additionalProperties": false,
+      }
+    end
+
+    it "validates a json body against a schema" do
+      expect(described_class.fully_validate(schema, body, schema_reader: JSON::Schema::Reader.new)).to eq []
+      expect(*described_class.fully_validate(schema, invalid_body, schema_reader: JSON::Schema::Reader.new)).to include("did not contain a required property of 'a'").once
+      expect(*described_class.fully_validate(schema, extra_attributes, schema_reader: JSON::Schema::Reader.new)).to include("contains additional properties [\"extra_stuff\"] outside of the schema when none are allowed in schema").once
```

Comment:
Is there some matcher that can help us avoid stringifying the output?

Something like `expect(described_class.fully_validate(schema, invalid_body, schema_reader: JSON::Schema::Reader.new).to contain_exactly("did not contain a required property of 'a'")`?

Reply:
I considered using a regex since matching is harder for this one since it generates a schema uuid as part of the original message:
`The property '#/' did not contain a required property of 'a' in schema 78d55aeb-99b3-565f-b95e-fdb2e1014a93"`
As a later step, I want to parse the string to extract what exactly was missing and feed that into the error message that's presented back, but for the test's stringifying and doing an include was the simplest thing that could work.

Reply:
Alright, that makes sense. Happy to keep string matching in that case :)